core/src/main/java/org/apache/calcite/sql/SqlCreateProcedure.java

```diff
@@ -105,6 +105,7 @@ public SqlCreateProcedure(SqlParserPos pos, SqlCreateSpecifier createSpecifier,
 
   @Override public void validate(SqlValidator validator,
       SqlValidatorScope scope) {
+    validator.addProcedureToSchema(this);
```

Comment:
Why not follow the pattern used for other nodes that call back to SqlValidatorImpl with something like `validate<NodeType>()`? In this case, that would be `validateCreateProcedure()`. That function could take care of adding the procedure to the schema and validating the underlying scripting statement.

Reply:
Sure, that makes more sense.